.github/workflows/unit_tests.yml

```diff
@@ -54,6 +54,12 @@ jobs:
       with:
         python-version: '3.9.x'
 
+    - name: Set up redis
+      run: |
+        sudo apt update -y
+        sudo apt install redis
+        sudo systemctl start redis
+
```

Comment:
Maybe a suggestion is to use a redis action?
Something like: https://github.com/marketplace/actions/actions-setup-redis
There are also others that might be better
Is maybe more stable and you can fix the versions

Reply:
It only has 19 stars on github. I think redis api will not change in future. If this gives trouble in future, we will change it.